### PR TITLE
VS project fixes

### PR DESCRIPTION
**Pull request by NicknineTheEagle**

### CODE REVIEW

**Comment by kcgen** [comment]
@NicknineTheEagle , the MSVC 64bit build is failing with:

``` text
LINK : fatal error LNK1181: cannot open input file 'zlibd.lib' [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
```
Maybe the debug lib is named something else (or doesn't exist?) Worst case, suggest rolling just this one back to the original lib name.

**Comment by NicknineTheEagle** [comment]
It compiles on my end while it *wasn't* compiling with the old name.

**Comment by NicknineTheEagle** [comment]
> @NicknineTheEagle , the MSVC 64bit build is failing with:
> 
> ```
> LINK : fatal error LNK1181: cannot 
> open input file 'zlibd.lib' 
> [D:\a\dosbox-staging\dosbox-staging
> \src\libs\zmbv\zmbv.vcxproj]
> ```
> 
> Maybe the debug lib is named something else (or doesn't exist?) Worst case, suggest rolling just this one back to the original lib name.

It's called zlib.lib on my end and zmbv *wasn't* compiling before the tweak.

**Comment by kcgen** [comment]
> It compiles on my end while it _wasn't_ compiling with the old name.

Sounds like there's a difference in the libs setup under CI; and/or maybe the setup documentation needs to be updated?

`.github/workflows/windows-msvc.yml` has the bootstrap and vcpkg steps used in CI. 

Curious if anyone else hits this (links errors when building `main` under MSVC), or know how we can fix this.

File: .github/workflows/windows-msvc.yml [hunk 1]
```diff
@@ -68,8 +68,7 @@ jobs:
         shell: pwsh
         run: |
           cd vs
-          # TODO: get unit tests working in Debug-mode; for now use release flags (better than nothing)
-          MSBuild -m dosbox.sln -t:tests:Rebuild -p:Configuration=Release -p:Platform=${{ matrix.conf.arch }}
+          MSBuild -m dosbox.sln -t:tests:Rebuild -p:Configuration=Debug -p:Platform=${{ matrix.conf.arch }}
```
Comment (kcgen) [inline .github/workflows/windows-msvc.yml:71]: :👍

Comment (NicknineTheEagle) [reply .github/workflows/windows-msvc.yml:71]: Unit tests appear to succeed in Debug but that doesn't fix the build issue we ran into earlier.

**Comment by NicknineTheEagle** [comment]
```
       VcpkgTripletSelection:
         Using triplet "x64-windows" from "D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\x64-windows\"
         Using normalized configuration "Release"
```
This looks suspicious - why does vcpkg fetch Release dependencies for Debug build?

**Comment by shermp** [comment]
@NicknineTheEagle I just tested compiling `main` with VS 2019 16.11.21, in x64 Debug configuration. It compiles just fine.

> ```
>        VcpkgTripletSelection:
>          Using triplet "x64-windows" from "D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\x64-windows\"
>          Using normalized configuration "Release"
> ```
> 
> This looks suspicious - why does vcpkg fetch Release dependencies for Debug build?

Check the `vcpkg_installed\<triplet>\<triplet>` dir. There should be `debug\lib` with the debug libraries. There is on mine. And I got the same `Using normalized configuration "Release"` message from memory (I did this a few days ago).

**Comment by NicknineTheEagle** [comment]
@shermp Yeah, like I said, Debug compiles fine on my end, it's just CI that has problems for some reason.

**Comment by NicknineTheEagle** [comment]
@shermp `main` refuses to build on my end with linking errors.

**Comment by shermp** [comment]
I setup my vcpkg a few days ago. I wonder if something broke or changed on their end?

**Comment by shermp** [comment]
FWIW, I just rebased and pushed my SLIRP branch, and the MSVC CI ran without errors. I'm not sure what's going on here!

**Comment by NicknineTheEagle** [comment]
> FWIW, I just rebased and pushed my SLIRP branch, and the MSVC CI ran without errors. I'm not sure what's going on here!

`main` currently builds on CI without issues but fails on my end. PR works for me but fails on CI. That's what's strange.

**Comment by shermp** [comment]
@NicknineTheEagle what is your Visual Studio version?

**Comment by NicknineTheEagle** [comment]
> @NicknineTheEagle what is your Visual Studio version?

VS 2022 with v142 toolset installed. Pretty sure CI is same.

**Comment by shermp** [comment]
This is probably a silly question, but did you run `vcpkg integrate install` during VCPKG installation? (After VS is installed)

**Comment by NicknineTheEagle** [comment]
> This is probably a silly question, but did you run `vcpkg integrate install` during VCPKG installation? (After VS is installed)

Of course.

**Comment by shermp** [comment]
With this branch, and my existing `vcpkg_installed` dir, I get the following error:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK1181	cannot open input file 'zlibd.lib'	zmbv	C:\msys64\home\sherm\dosbox-staging\src\libs\zmbv\LINK	1	
```

**Comment by kcgen** [comment]
@NicknineTheEagle , I looked into `main`'s last MVSC CI runs, and both vcpkg's tree are build with release flags (which you caught and switched to debug here in this PR).

Here are those relevant snippets:

### 64-bit non-release workflow
 - Unit tests: `Building solution configuration "Release|x64"`
 - Executable + ZMBV: `Building solution configuration "Debug|x64"`

https://github.com/dosbox-staging/dosbox-staging/commit/eb248c342a8f1fb592ce226ad379e0573b41cd3e/checks/10422663039/logs

```text
2023-01-03T21:44:28.8467469Z        VcpkgTripletSelection:
2023-01-03T21:44:28.8469516Z          Using triplet "x64-windows" from "D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\x64-windows\"
2023-01-03T21:44:28.8470136Z          Using normalized configuration "Release"
2023-01-03T21:44:28.8473941Z        VcpkgInstallManifestDependencies:
2023-01-03T21:44:28.8475384Z          Installing vcpkg dependencies to D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\
2023-01-03T21:44:28.8476281Z          Creating directory "D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\".
2023-01-03T21:44:28.8479710Z          "C:\vcpkg\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows" --vcpkg-root "C:\vcpkg\\" "--x-manifest-root=D:\a\dosbox-staging\dosbox-staging\\" "--x-install-root=D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\\" 
2023-01-03T21:44:29.0512209Z          "C:\vcpkg\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows" --vcpkg-root "C:\vcpkg\\" "--x-manifest-root=D:\a\dosbox-staging\dosbox-staging\\" "--x-install-root=D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\\" 
2023-01-03T21:44:32.1188859Z          Detecting compiler hash for triplet x64-windows...
2023-01-03T21:44:47.3111422Z          The following packages will be built and installed:
2023-01-03T21:44:47.3112901Z            * bzip2[core,tool]:x64-windows -> 1.0.8#3
2023-01-03T21:44:47.3114261Z            * dirent[core]:x64-windows -> 1.23.2#1
2023-01-03T21:44:47.3115998Z              fluidsynth[buildtools,core]:x64-windows -> 2.2.8
2023-01-03T21:44:47.3116504Z            * gettext[core]:x64-windows -> 0.21#9
2023-01-03T21:44:47.3116973Z            * glib[core]:x64-windows -> 2.75.0
2023-01-03T21:44:47.3117404Z              gtest[core]:x64-windows -> 1.12.1
2023-01-03T21:44:47.3117930Z              iir1[core]:x64-windows -> 1.9.1
2023-01-03T21:44:47.3118923Z            * libffi[core]:x64-windows -> 3.4.4
2023-01-03T21:44:47.3119304Z            * libiconv[core]:x64-windows -> 1.17
2023-01-03T21:44:47.3119665Z              libmt32emu[core]:x64-windows -> 2.7.0
2023-01-03T21:44:47.3120041Z            * libogg[core]:x64-windows -> 1.3.5
2023-01-03T21:44:47.3120419Z              libpng[core]:x64-windows -> 1.6.39
2023-01-03T21:44:47.3122000Z            * opus[core]:x64-windows -> 1.3.1#9
2023-01-03T21:44:47.3122479Z              opusfile[core]:x64-windows -> 0.12#1
2023-01-03T21:44:47.3122858Z            * pcre2[core]:x64-windows -> 10.40#1
2023-01-03T21:44:47.3123239Z              sdl2[base,core]:x64-windows -> 2.26.1#1
2023-01-03T21:44:47.3123602Z              sdl2-image[core]:x64-windows -> 2.6.2#1
2023-01-03T21:44:47.3123990Z              sdl2-net[core]:x64-windows -> 2.2.0#1
2023-01-03T21:44:47.3124371Z              speexdsp[core]:x64-windows -> 1.2.1
2023-01-03T21:44:47.3124717Z              tracy[core]:x64-windows -> 0.9.0
2023-01-03T21:44:47.3125119Z            * vcpkg-cmake[core]:x64-windows -> 2022-10-30
2023-01-03T21:44:47.3125565Z            * vcpkg-cmake-config[core]:x64-windows -> 2022-02-06#1
2023-01-03T21:44:47.3125974Z            * vcpkg-tool-meson[core]:x64-windows -> 0.63
2023-01-03T21:44:47.3126354Z              zlib[core]:x64-windows -> 1.2.13
```

### 64-bit release workflow
 - Executable + ZMBV: `Building solution configuration "Release|x64"`

https://github.com/dosbox-staging/dosbox-staging/commit/eb248c342a8f1fb592ce226ad379e0573b41cd3e/checks/10422663290/logs

```text
2023-01-03T21:43:48.8361660Z        VcpkgTripletSelection:
2023-01-03T21:43:48.8363485Z          Using triplet "x64-windows" from "D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\x64-windows\"
2023-01-03T21:43:48.8364128Z          Using normalized configuration "Release"
2023-01-03T21:43:48.8366806Z        VcpkgInstallManifestDependencies:
2023-01-03T21:43:48.8367684Z          Installing vcpkg dependencies to D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\
2023-01-03T21:43:48.8368223Z          Creating directory "D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\".
2023-01-03T21:43:48.8370279Z          "C:\vcpkg\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows" --vcpkg-root "C:\vcpkg\\" "--x-manifest-root=D:\a\dosbox-staging\dosbox-staging\\" "--x-install-root=D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\\" 
2023-01-03T21:43:48.8864413Z          "C:\vcpkg\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows" --vcpkg-root "C:\vcpkg\\" "--x-manifest-root=D:\a\dosbox-staging\dosbox-staging\\" "--x-install-root=D:\a\dosbox-staging\dosbox-staging\vcpkg_installed\x64-windows\\" 
2023-01-03T21:43:52.6086112Z          Detecting compiler hash for triplet x64-windows...
2023-01-03T21:44:14.3178577Z          The following packages will be built and installed:
2023-01-03T21:44:14.3179555Z            * bzip2[core,tool]:x64-windows -> 1.0.8#3
2023-01-03T21:44:14.3181033Z            * dirent[core]:x64-windows -> 1.23.2#1
2023-01-03T21:44:14.3181463Z              fluidsynth[buildtools,core]:x64-windows -> 2.2.8
2023-01-03T21:44:14.3181825Z            * gettext[core]:x64-windows -> 0.21#9
2023-01-03T21:44:14.3182139Z            * glib[core]:x64-windows -> 2.75.0
2023-01-03T21:44:14.3182355Z              gtest[core]:x64-windows -> 1.12.1
2023-01-03T21:44:14.3182582Z              iir1[core]:x64-windows -> 1.9.1
2023-01-03T21:44:14.3183390Z            * libffi[core]:x64-windows -> 3.4.4
2023-01-03T21:44:14.3183644Z            * libiconv[core]:x64-windows -> 1.17
2023-01-03T21:44:14.3185737Z              libmt32emu[core]:x64-windows -> 2.7.0
2023-01-03T21:44:14.3186864Z            * libogg[core]:x64-windows -> 1.3.5
2023-01-03T21:44:14.3187156Z              libpng[core]:x64-windows -> 1.6.39
2023-01-03T21:44:14.3187395Z            * opus[core]:x64-windows -> 1.3.1#9
2023-01-03T21:44:14.3187626Z              opusfile[core]:x64-windows -> 0.12#1
2023-01-03T21:44:14.3187871Z            * pcre2[core]:x64-windows -> 10.40#1
2023-01-03T21:44:14.3188107Z              sdl2[base,core]:x64-windows -> 2.26.1#1
2023-01-03T21:44:14.3188366Z              sdl2-image[core]:x64-windows -> 2.6.2#1
2023-01-03T21:44:14.3188703Z              sdl2-net[core]:x64-windows -> 2.2.0#1
2023-01-03T21:44:14.3188931Z              speexdsp[core]:x64-windows -> 1.2.1
2023-01-03T21:44:14.3189165Z              tracy[core]:x64-windows -> 0.9.0
2023-01-03T21:44:14.3189409Z            * vcpkg-cmake[core]:x64-windows -> 2022-10-30
2023-01-03T21:44:14.3189674Z            * vcpkg-cmake-config[core]:x64-windows -> 2022-02-06#1
2023-01-03T21:44:14.3189948Z            * vcpkg-tool-meson[core]:x64-windows -> 0.63
2023-01-03T21:44:14.3190183Z              zlib[core]:x64-windows -> 1.2.13
```

I guess you might have to take a listing of your vcpkg directory tree, and compare it's libs to those that are produced after these CI jobs?  You could insert some of those recurive DIR commands in the YAML, to get some visibility.

@shermp , if you did a fresh vcpkg install, I'm guessing you just followed the docs <https://github.com/dosbox-staging/dosbox-staging/blob/main/README.md#windows---visual-studio-2019-or-newer> (which I think you updated when you moved from the old package-name-list style to the new manifest-based approach).

**Comment by shermp** [comment]
If I set `Configuration Properties` > `Advanced` > `Use Debug Libraries` AND change the linker to use `zlibd.lib` on main, I can compile with the debug zlib library. I need both for it to work though. This seems to behave like @NicknineTheEagle 's setup, even though that `Use Debug Libraries` option is set to `No` in the project.

Stranger and stranger...

**Comment by shermp** [comment]
Ok, lets forget the `Use Debug Libraries` option for now. There is actually a `vcpkg` specific option to set for using debug libraries.

@NicknineTheEagle can you try updating this PR by setting `Configuration Properties` > `vcpkg` > `Vcpkg configuration` to `Debug` in all projects? See if that compiles for you and CI.

**Comment by LowLevelMahn** [comment]
i've made a clean clone of vcpkg + fresh clone of dosbox
builds under Win10, VS2019 latestest for me (in Debug)

**Comment by kcgen** [comment]
> I guess the question is, do we want to link debug builds against debug 3rd party libraries? If so, this PR will still be useful (with the added vcpkg config option).

I think it makes the most sense, if it can be made to work. At least it's an easy way to make sure all the debug libs are present and linkable.

**Comment by kcgen** [comment]
Any potential to unjam this, or it is stuck in a stalemate?

With the existing `main` VS package files running in CI and with both @LowLevelMahn and @shermp reporting that the VS packages build locally, I suspect a the cause is something in your vcpkg configuration, @NicknineTheEagle.

Now, a separate topic is if we want to move to all debug libs for the unit test/debug builds - which would be nice to do, regardless of this other issue.

File: vs/dosbox.vcxproj [hunk 2]
```diff
@@ -169,7 +173,7 @@
       <ObjectFileName>$(IntDir)\$(ProjectName)\$(ConfigurationName)\%(RelativeDir)</ObjectFileName>
     </ClCompile>
     <Link>
-      <AdditionalDependencies>opengl32.lib;SDL2_image.lib;SDL2_net.lib;winmm.lib;libpng16.lib;SDL2main.lib;SDL2.lib;odbc32.lib;odbccp32.lib;ws2_32.lib;mt32emu.lib;opusfile.lib;opus.lib;ogg.lib;fluidsynth.lib;speexdsp.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>opengl32.lib;SDL2_imaged.lib;SDL2_netd.lib;winmm.lib;libpng16d.lib;SDL2maind.lib;SDL2d.lib;odbc32.lib;odbccp32.lib;ws2_32.lib;mt32emu.lib;opusfile.lib;opus.lib;ogg.lib;fluidsynth.lib;speexdsp.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>$(OutDir)\dosbox.exe</OutputFile>
```
Comment (kcgen) [inline vs/dosbox.vcxproj:177]: Suggest a more specific commit message: `Fix VS Debug build` implies the debug build is broken, which it isn't: a binaries with DOSBox Staging source code is indeed compiled with debug instrumentation and is passing under CI.

File: .github/workflows/windows-msvc.yml (hunk 1 above)
Comment (kcgen) [inline .github/workflows/windows-msvc.yml:71]: `Use Debug config for unit tests in CI MSVC Debug job`

:+1: ; great commit message.

File: vs/dosbox.vcxproj (hunk 2 above)
Comment (NicknineTheEagle) [reply vs/dosbox.vcxproj:177]: "Explicitly use Debug dependencies in VS Debug config".

**Review by kcgen — APPROVED** [review]
Thanks, @NicknineTheEagle.

**Pull request merged**